lib/modules/datasource/cpan/index.ts

```diff
@@ -0,0 +1,147 @@
+let contentLength = 0;
+
+const pathExtensionPattern = regEx(/(\.tgz|\.tar\.gz)$/);
+const pathPattern = regEx(/^.+\/(.+)-(?:\d+(?:\.\d+)*)(?:\.tgz|\.tar\.gz)$/);
+
+// Note: use only for tests
+export function resetCache(): void {
+  lastSync = new Date('2000-01-01');
+  packages = Object.create(null);
+  contentLength = 0;
+}
+
+export class CpanDatasource extends Datasource {
+  static readonly id = 'cpan';
+
+  constructor() {
+    super(CpanDatasource.id);
+  }
+
+  override readonly defaultRegistryUrls = ['https://www.cpan.org'];
+
+  override readonly defaultVersioning = perlVersioning.id;
+
+  @cache({
+    namespace: `datasource-${CpanDatasource.id}`,
+    key: ({ packageName }: GetReleasesConfig) => `${packageName}`,
+  })
+  override async getReleases({
+    packageName,
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    // istanbul ignore if
+    if (!registryUrl) {
+      return null;
+    }
+
+    await this.syncVersions(registryUrl);
+    const pkg = packages[packageName];
+    if (!pkg) {
+      return null;
+    }
+    const dep: ReleaseResult = {
+      releases: [pkg.release],
+      changelogUrl: `https://metacpan.org/dist/${pkg.distribution}/changes`,
+      homepage: `https://metacpan.org/pod/${packageName}`,
+    };
+    return dep;
+  }
+
+  async updateCpanVersions(registryUrl: string): Promise<void> {
+    const url = joinUrlParts(registryUrl, 'modules', '02packages.details.txt');
+    const options = {
+      headers: {
+        'accept-encoding': 'identity',
+        range: `bytes=${contentLength}-`,
+      },
+    };
+    let newLines: string;
+    try {
+      logger.trace({ registryUrl }, 'Fetching 02packages.details.txt');
+      const startTime = Date.now();
+      newLines = (await this.http.get(url, options)).body;
+      const durationMs = Math.round(Date.now() - startTime);
+      logger.trace(
+        { registryUrl, durationMs },
+        'Fetched 02packages.details.txt'
+      );
+    } catch (err) /* istanbul ignore next */ {
+      if (err.statusCode !== 416) {
+        contentLength = 0;
+        packages = Object.create(null); // Because we might need a "constructor" key
+        logger.trace({ registryUrl, err }, 'fetch error');
+        throw new ExternalHostError(new Error('CPAN fetch error'));
+      }
+      logger.trace({ registryUrl }, 'No update');
+      lastSync = new Date();
+      return;
+    }
+
+    for (const line of newLines.split(newlineRegex)) {
+      CpanDatasource.processLine(line);
+    }
+    lastSync = new Date();
+  }
+
+  private static processLine(line: string): void {
+    let split: string[] | undefined;
+    let pkg: string | undefined;
+    let latestVersion: string | undefined;
+    let path: string | undefined;
+    try {
+      const l = line.trim();
+      if (!l.length || !pathExtensionPattern.test(l)) {
+        return;
+      }
+      split = l.split(/\s+/);
+      [pkg, latestVersion, path] = split;
+      pkg = copystr(pkg);
```

Comment:
are you sure we need to copy the string here? if yes, you also need to copy `latestVersion`. 🤔